kola/tests/docker/docker_util.go

```diff
@@ -0,0 +1,167 @@
+
+func registerDockerTests(t dockerTest) {
+	targets := t.targets
+	if len(targets) == 0 {
+		targets = []dockerTestTarget{dockerTestTargetSkim1126, dockerTestTargetHost}
+	}
+	for _, target := range targets {
+		targetMap[target](t)
+	}
+}
+
+// targetMap specifies functions that will take a generic dockerTest and
+// transform it into a kola test for the given target.
+var targetMap = map[dockerTestTarget]dockerTestRegistrationFn{
+	dockerTestTargetHost:     targetHostDocker,
+	dockerTestTargetSkim1126: targetDockerSkim1126,
+}
+
+// dockerTest specifies the reusable portions of a test, and which docker
+// targets it may be run against.
+type dockerTest struct {
+	// name is the name of the test. It maps to a kola name, but will
+	// automatically be prefixed with a target-specific string (e.g. `docker.` +
+	// name).
+	name string
+	// test specifies the test to register.
+	// Note that the 'name', 'userdata', and 'platform' variables may be modified
+	// by a given registration function
+	test register.Test
+	// targets, if set, will choose the specific docker targets for this test
+	// from the above list.  If unset or empty, all targets will be run
+	// Only ignition config is supported by the docker skim target.
+	targets []dockerTestTarget
+}
+
+// targetDockerSkim1126 transforms a test to run on docker skim
+func targetDockerSkim1126(t dockerTest) {
+	name := t.name
+	test := &t.test
+	currentPlatforms := test.Platforms
+	dockerSkimPlatforms := []string{"aws", "gce"}
+
+	// Filter for the set of supported platforms from those the caller chose
+	supportedSpecifiedPlatforms := currentPlatforms[:0]
+	for _, platform := range currentPlatforms {
+		for _, supported := range dockerSkimPlatforms {
+			if supported == platform {
+				supportedSpecifiedPlatforms = append(supportedSpecifiedPlatforms, platform)
+				break
+			}
+		}
+	}
+
+	if len(currentPlatforms) == 0 {
+		// Default to the full supported set
+		test.Platforms = dockerSkimPlatforms
+	} else {
+		test.Platforms = supportedSpecifiedPlatforms
+	}
+
+	test.Name = fmt.Sprintf("%s.%s", dockerTestTargetSkim1126, name)
+
+	ign := types.Config{
+		Ignition: types.Ignition{
+			Version: types.IgnitionVersion(semver.Version{
+				Major: 2,
+				Minor: 0,
+				Patch: 0,
+			}),
+		},
+	}
+	if test.UserData != "" {
+		var err error
+		ign, err = config.Parse([]byte(test.UserData))
+		if err != nil {
+			plog.Fatalf("invalid ignition config for %v: %v", test.Name, err)
+		}
+	}
+
+	skimUnit := types.SystemdUnit{
+		Name: "docker.service",
+		// https://github.com/coreos/docker-skim/blob/master/Documentation/using-docker-skim.md
+		Contents: `
+[Unit]
+Description=Docker Application Container Engine
+Documentation=http://docs.docker.com
+After=docker.socket
+Requires=docker.socket
+
+[Service]
+Type=simple
+
+ExecStart=/usr/bin/rkt run --dns=host --interactive \
+  --trust-keys-from-https \
+  --stage1-name=users.developer.core-os.net/skim/stage1-skim:0.0.1 \
```

Comment:
In the past I've posted source tarballs under `gs://coreos-net-storage/mirror/snapshots/` (backend for `https://storage.core-os.net/mirror/snapshots/`). So some path under the coreos-net-storage may be appropriate. And given the --trust-keys-from-https that seems better too since the ssl connection between CF and Google is strictly enforced for that bucket while it is not for the users bucket (we would have had to rename the users bucket to have robust ssl)